src/org/zaproxy/zap/extension/jsonView/HttpPanelJsonView.java

```diff
@@ -0,0 +1,185 @@
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.jsonView;
+
+import java.awt.BorderLayout;
+import java.util.regex.Pattern;
+
+import javax.swing.*;
+
+import net.sf.json.*;
+import org.apache.commons.configuration.FileConfiguration;
+import org.fife.ui.rtextarea.RTextScrollPane;
+import org.parosproxy.paros.network.HttpBody;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.httppanel.Message;
+import org.zaproxy.zap.extension.httppanel.view.*;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestBodyStringHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseBodyStringHttpPanelViewModel;
+
+public class HttpPanelJsonView implements HttpPanelView, HttpPanelViewModelListener {
+
+	/**
+	 * Default name used for {@code MessageContainer}.
+	 *
+	 * @see org.zaproxy.zap.view.messagecontainer.MessageContainer
+	 */
+	public static final String DEFAULT_MESSAGE_CONTAINER_NAME = "HttpMessagePanel";
+	public static final String NAME = "HttpPanelJsonView";
+	private static final String CAPTION_NAME = "Json";
+
+	private JTextArea httpPanelJsonArea;
+	private JPanel mainPanel;
+	private AbstractStringHttpPanelViewModel model;
+	private Pattern quickcheck;
+
+	public HttpPanelJsonView(AbstractStringHttpPanelViewModel model) {
+		httpPanelJsonArea = new HttpPanelJsonArea();
+		RTextScrollPane scrollPane = new RTextScrollPane(httpPanelJsonArea);
+		scrollPane.setLineNumbersEnabled(false);
+		mainPanel = new JPanel(new BorderLayout());
+		mainPanel.add(scrollPane, BorderLayout.CENTER);
+		this.model = model;
+		model.addHttpPanelViewModelListener(this);
+
+		//a very liberal regex for enabling the view!
+		quickcheck = Pattern.compile("\\A\\s*[{\\[\"].*[}\\]\"]\\s*\\z", Pattern.DOTALL);
+	}
+
+	@Override
+	public void setSelected(boolean selected) {
+		if (selected) {
+			httpPanelJsonArea.requestFocusInWindow();
+		}
+	}
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public String getCaptionName() {
+		return CAPTION_NAME;
+	}
+
+	@Override
+	public String getTargetViewName() {
+		return "";
+	}
+
+	@Override
+	public int getPosition() {
+		return 0;
+	}
+
+	@Override
+	public boolean isEnabled(Message message) {
+		if(message == null){

```

Comment:
Could do the `instanceof` check here to return immediately (if it was passed other `Message` type it would result in a NPE, `jsonString` is `null` in that case).